Tasks/CopyFilesOverSSHV0/copyfilesoverssh.ts

```diff
@@ -191,34 +191,42 @@ async function run() {
             tl.debug('Number of files to copy = ' + filesToCopy.length);
             tl.debug('filesToCopy = ' + filesToCopy);
 
+            let failureCount = 0;
             tl._writeLine(tl.loc('CopyingFiles', filesToCopy.length));
-            var fileCopyProgress:Q.Promise<string> [] = [];
             for (var i:number = 0; i < filesToCopy.length; i++) {
-                var fileToCopy = filesToCopy[i];
-                tl.debug('fileToCopy = ' + fileToCopy);
-
-                var relativePath;
-                if (flattenFolders) {
-                    relativePath = path.basename(fileToCopy);
-                } else {
-                    relativePath = fileToCopy.substring(sourceFolder.length)
-                        .replace(/^\\/g, "")
-                        .replace(/^\//g, "");
-                }
-                tl.debug('relativePath = ' + relativePath);
-                var targetPath = path.posix.join(targetFolder, relativePath);
-
-                tl._writeLine(tl.loc('StartedFileCopy', fileToCopy, targetPath));
-                if (!overwrite) {
-                    var fileExists:boolean = await sshHelper.checkRemotePathExists(targetPath);
-                    if (fileExists) {
-                        throw tl.loc('FileExists', targetPath);
+                try {
```

Comment:
I just added the try-catch inside the loop.

Reply:
So, we're okay with 2000 try/failures for 2000 files?

Reply:
Are you concerned about perf or something else?

Reply:
I guess a combination of perf and expectations. When a user sees a 2000 failures, are they going to wonder why we kept trying?

Reply:
To me, attempting to every file seems more reasonable than giving up on a random failure.  I think that's why users are complaining.  It's also more helpful in trying to diagnose the error which files work and which don't.

As for perf, I think it's hard to reason about -- depends on what the error is, size of the files, etc. (it could very well be faster to bail before going over the line on 2000 files as opposed to actually copying them).

At any rate, if a file fails to copy, the build is already doomed to fail and the user can just cancel if they're checking their watch.